openquake/hazardlib/calc/filters.py

```diff
@@ -157,6 +157,9 @@ def get_distances(rupture, mesh, param):
         dist = rupture.get_cdppvalue(mesh)
     elif param == 'azimuth':
         dist = rupture.surface.get_azimuth(mesh)
+    elif param == "rvolc":
```

Comment:
Yes, I'm not totally clear on what is the best way to represent this distance in the calculation. The option that seems most faithful to the intent of the author is to provide some sort of supplementary file into the calculation within which the volcanic zones are defined as a set of one or more polygons (the path to which could be input as an attribute of the GMPE in the logic tree file. Shapely can handle the intersection between lines and polygons but I'm concerned that this might affect the speed of the calculation.